lib/NormalModule.js

```diff
@@ -125,15 +129,53 @@ const asBuffer = input => {
 	return input;
 };
 
+/**
+ * @param {InputFileSystem} fs fs
+ * @returns {(resource: string, cb: (err: Error|null, data?: string|Buffer) => void) => void} read function
+ */
 const readResourceFn = fs => {
 	return (resource, callback) => {
-		const decodedData = decodeDataURI(resource);
+		const scheme = getScheme(resource);
+		if (scheme && isSchemeSupported(scheme)) {
+			switch (scheme) {
+				case "data":
+					return process.nextTick(() => {
+						callback(null, decodeDataURI(resource));
+					});
+				case "file":
+					return fs.readFile(fileURLToPath(new URL(resource)), callback);
+				case "http":
+					return http.get(new URL(resource), res => {
```

Comment:
`new URL(...)` is explicit to validate specifier.
should we add webpack header?
should we add timeout?

Reply:
I always thought that we should implement this at `enhanced-resolve` side (like plugins), not only js can have `http`/`https`/etc protcols, CSS and HTML also, even sass/less/etc

Reply:
I think we need to think about it better. Because it is a very important for the future design

Reply:
am.. this will be processed by loaders, take a look on test case https://github.com/webpack/webpack/pull/11095/files#diff-645b199fca61ee6e6a314b51cf63dc55R8

Reply:
Not all loaders convert `url()`/`<tag src="" />`/`@import`/etc stuff to `require`/`import`, therefore support for this only for built-in modules may be a little problematic in the future. Or maybe I misunderstand something here and think in the wrong direction

Reply:
`loaderContext. loadModule` should work. Resolver can help in resolving, e.g. trying N ending to `http://example.com/module`, but this violating spec (as I understand) since native browser import does not do any extra resolving as well as Node.js.